src/domain/metadata/test/retrieveDataverseItemMetadata.spec.ts

```diff
@@ -0,0 +1,218 @@
+    { property: 'format', value: 'Shapefile' }
+  ],
+  isLoading: false
+}
+const initStore = (initialState?: MetadataDomain.Options): StoreApi<MetadataDomain.DomainAPI> =>
+  MetadataDomain.storeFactory(initialState)
+
+// eslint-disable-next-line max-lines-per-function
+describe(`Retrieve metadata from a dataverve item by its ID`, () => {
+  describe.each`
+    preloadedState                        | expectedState
+    ${undefined}                          | ${defaultState}
+    ${preloadedStateWithAuditMetadata1}   | ${stateWithAuditMetadata1}
+    ${preloadedStateWithGeneralMetadata1} | ${stateWithGeneralMetadata1}
+    ${preloadedStateWithLoading1}         | ${stateWithLoading1}
+  `('Given a preloaded state <$preloadedState>', ({ preloadedState, expectedState }: Data1) => {
+    describe(`when initializing state`, () => {
+      const store = initStore(preloadedState)
+
+      test(`then expect state to be ${JSON.stringify(expectedState)}`, () => {
+        const currentState: MetadataState = {
+          auditMetadata: store.getState().auditMetadata()(),
+          generalMetadata: store.getState().generalMetadata()(),
+          isLoading: store.getState().isLoading()()
+        }
+        expect(currentState).toStrictEqual(expectedState)
+      })
+    })
+  })
+
+  describe.each`
+    preloadedState                       | dataverseItemId | mockGatewayResult      | expectedState                        | error
+    ${undefined}                         | ${''}           | ${gatewayResult1}      | ${defaultState}                      | ${undefined}
+    ${undefined}                         | ${'id1'}        | ${gatewayResult2}      | ${stateWithGeneralAndAuditMetadata1} | ${undefined}
+    ${stateWithGeneralAndAuditMetadata1} | ${'id1'}        | ${gatewayResult3}      | ${stateWithGeneralAndAuditMetadata2} | ${undefined}
+    ${stateWithGeneralAndAuditMetadata1} | ${'id1'}        | ${gatewayResult3}      | ${stateWithGeneralAndAuditMetadata2} | ${undefined}
+    ${undefined}                         | ${'id1'}        | ${gatewayErrorResult1} | ${defaultState}                      | ${NetworkUnspecifiedError('Network crashed')}
+    ${undefined}                         | ${'id1'}        | ${gatewayErrorResult2} | ${defaultState}                      | ${SerializationError('Serialization failed')}
+  `(
+    'Given a preloaded state <$preloadedState> and a dataverse item id <$dataverseItemId> with this context dependency <$deps>',
+    ({ preloadedState, dataverseItemId, mockGatewayResult, expectedState, error }: Data2) => {
+      describe(`When calling retrieveDataverseItemMetadata command `, () => {
+        afterEach(() => {
+          jest.restoreAllMocks()
+        })
+
+        const store = initStore(preloadedState)
+
+        test(`then expect current state to be ${JSON.stringify(expectedState)}`, async () => {
+          const spiedMetadataGateway = jest.spyOn(metadataSparqlGateway, 'retrieveMetadata')
```

Comment:
Should be declared at upper `describe` level with the `jest.restoreAllMocks()` function.